rust/core-lib/src/annotations.rs

```diff
@@ -44,14 +43,20 @@ impl AnnotationType {
 pub type AnnotationRange = Vec<[usize; 4]>;
```

Comment:
seeing this in use below now i wonder if it might make sense to make this a struct or something with named fields? Just thinking out loud. 🤷‍♂️

Reply:
Yeah, it's more readable as a struct